fix(LabelMaps): Support loading and displaying annotations

scottwittenburg opened the pull request:
Supports loading annotations via url parameter `annotations`.  Example annotations file can be found [here](https://thewtex.github.io/allen-ccf-itk-vtk-zarr/allen_ontology_50.json).

scottwittenburg commented:
@thewtex, while this is basically working end-to-end, I think we'll need to iterate on it a little.  For example:

1. I can't remember if the label values correspond to the index in, e.g. `allen_ontology_50.json`, or if it's the `id` values.
2. I show the 'acronym` field, rather than the `name` field, because of how long the `name` seems to be.
3. This does not yet support drag-and-drop annotations files.

scottwittenburg commented on src/ItkVtkViewProxy.js:
```diff
@@ -93,6 +93,14 @@ function ItkVtkViewProxy(publicAPI, model) {
   }
 
+  function getAnnotationText(value) {
+    const labelValue = value[model.labelIndex]
```
Should this function live here and the `annotationMap` and `labelIndex` provided to this thing's `model`?  Or should we just provide this proxy class with a method that can be called to retrieve the annotation text?

thewtex replied:
This looks good to me :+1:

thewtex commented:
Looking good, thanks @scottwittenburg! I will take a closer look later on today.

thewtex commented on src/ItkVtkViewProxy.js:
```diff
@@ -10,7 +10,7 @@ import vtkWidgetManager from 'vtk.js/Sources/Widgets/Core/WidgetManager'
 
 const CursorCornerAnnotation =
-  '<table class="corner-annotation" style="margin-left: 0;"><tr><td style="margin-left: auto; margin-right: 0;">Index:</td><td>${iIndex},</td><td>${jIndex},</td><td>${kIndex}</td></tr><tr><td style="margin-left: auto; margin-right: 0;">Position:</td><td>${xPosition},</td><td>${yPosition},</td><td>${zPosition}</td></tr><tr><td style="margin-left: auto; margin-right: 0;"">Value:</td><td>${value}</t ...
+  '<table class="corner-annotation" style="margin-left: 0;"><tr><td style="margin-left: auto; margin-right: 0;">Index:</td><td>${iIndex},</td><td>${jIndex},</td><td>${kIndex}</td></tr><tr><td style="margin-left: auto; margin-right: 0;">Position:</td><td>${xPosition},</td><td>${yPosition},</td><td>${zPosition}</td></tr><tr><td style="margin-left: auto; margin-right: 0;"">Value:</td><td>${value}</t ...
```
Can the *Annotation* row's presence be conditional on the presence of a label map?

thewtex commented on src/ItkVtkViewProxy.js:
```diff
@@ -93,6 +93,14 @@ function ItkVtkViewProxy(publicAPI, model) {
+    if (model.annotationMap != null && model.annotationMap.has(labelValue)) {
+      return model.annotationMap.get(labelValue)
+    }
+    return 'Unknown'
```
`labelValue` string is preferred.

thewtex commented on src/createViewer.js:
```diff
@@ -23,6 +23,7 @@ const createViewer = (
     multiscaleImage,
     labelMap,
     multiscaleLabelMap,
+    annotations,
```
-> `labelMapAnnotations` since geometries could have lookup tables.

thewtex commented on src/processFiles.js:
```diff
@@ -176,6 +187,14 @@ export const readFiles = async ({
+      annotationData = new Map(
+        annotations
+          .filter(item => !!item)
+          .map(({ acronym }, idx) => [idx, acronym])
```
-> factor out for generality.

scottwittenburg commented:
@thewtex Mind taking another look when you get a chance?  I think I addressed all the concerns you raised, and if we can get it merged, I would like to make use of some of this work in #282.

Thanks!

thewtex commented:
:tada: This PR is included in version 9.24.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/itk-vtk-viewer/v/9.24.0)
- [GitHub release](https://github.com/Kitware/itk-vtk-viewer/releases/tag/v9.24.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
